go/tasks/plugins/array/catalog.go

```diff
@@ -121,6 +121,13 @@ func DetermineDiscoverability(ctx context.Context, tCtx core.TaskExecutionContex
 	}
 
 	// Otherwise, run the data catalog steps - create and submit work items to the catalog processor,
+	cfg := catalog.GetConfig()
+	if int(arrayJobSize) > cfg.WriterWorkqueueConfig.IndexCacheMaxItems || int(arrayJobSize) > cfg.ReaderWorkqueueConfig.IndexCacheMaxItems {
```

Comment:
Do we not have a check on the array job size at an earlier point? Is it not sufficient? Should we update it to be so we fail as early as possible

Reply:
Great point. The [current check ](https://github.com/flyteorg/flyteplugins/blob/cdfc191012c2b2d5e3d956dec031df790b75386e/go/tasks/plugins/array/k8s/management.go#L69-L74) is performed to check if the execution array size (subtasks that are not cache hits) is larger than the max configured array job size. Two things (1) the check added in this PR does actually happen before the existing check (perhaps we should move the existing to perform at the same time as this one?) and (2) the checks have some difference, the existing check is on the number of executed subtasks and this new check pertains only the the cached items. In diving into the code we have a separate configuration for the size of the LRU cache for performing subtask cache lookups. The cache lookup only completes once all of the items are completed (available in the LRU cache). So intuitively, if we have an array size larger than the size of the LRU cache we will be stuck in an infinite loop.

I can move the existing check if that makes more sense? I think it probably does.